sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -703,17 +721,18 @@ private void downloadToFileCleanup(AsynchronousFileChannel channel) {
      *
      * @param metadata {@link Metadata}
      * @param accessConditions {@link BlobAccessConditions}
-     * @return A reactive response containing the ID of the new snapshot.
+     * @return A response containing a {@link BlobAsyncClient} which is used to interact with the created snapshot, use
+     * {@link BlobAsyncClient#getSnapshotId()} to get the identifier for the snapshot.
      */
-    public Mono<Response<String>> createSnapshot(Metadata metadata, BlobAccessConditions accessConditions) {
+    public Mono<Response<BlobAsyncClient>> createSnapshot(Metadata metadata, BlobAccessConditions accessConditions) {
         metadata = metadata == null ? new Metadata() : metadata;
         accessConditions = accessConditions == null ? new BlobAccessConditions() : accessConditions;
 
         return postProcessResponse(this.azureBlobStorage.blobs().createSnapshotWithRestResponseAsync(
             null, null, null, metadata, null, null,
             null, null, accessConditions.modifiedAccessConditions(),
             accessConditions.leaseAccessConditions(), Context.NONE))
-            .map(rb -> new SimpleResponse<>(rb, rb.deserializedHeaders().snapshot()));
+            .map(rb -> new SimpleResponse<>(rb, this.getSnapshotClient(rb.deserializedHeaders().snapshot())));
```

Comment:
Not very important but a thought, can we have getSnapshotId API?

Reply:
There is a getSnapshotId method in the client